### PR TITLE
change logs for AvP

### PR DESCRIPTION
**Pull request by emina-str**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/Avp.java [hunk 1]
```diff
@@ -248,25 +252,29 @@ public void addEmployment(EmploymentEntry employment) {
         Condition.isTrue(this.hasEmployment, "Can't add employment when flag is false or null!");
         Condition.notNull(employment);
         checkNotClosed();
+        ChangeLog changeLog = new ChangeLog();
         if (!findExistingEmployment(employment.getId()).isPresent()) {
             addRequiredDocument(employment.getId(), EMPLOYMENT);
         }
         this.employments.remove(employment);
         this.employments.add(employment);
+        changeLog.addChange(EMPLOYMENT_KEY, null, employment);
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/Avp.java:261]: what if we have already an employment?
then the values has changed.
i would look for the existing employment. if it doesn't exists, then add requirementDocument and add new employment in changelog, else add old and new employment in changelog.

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/AbsenceEntryRemovedEvent.java [hunk 2]
```diff
@@ -1,20 +1,13 @@
...
-    private final AbsenceEntry absence;
-
-    public AbsenceEntryRemovedEvent(Avp avp, AbsenceEntry absence) {
-        super(avp, new DomainEventType(AbsenceEntryRemovedEvent.class));
-        this.absence = absence;
-    }
-
-    public AbsenceEntry getAbsence() {
-        return absence;
+    public AbsenceEntryRemovedEvent(Avp avp, ChangeLog changeLog) {
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/AbsenceEntryRemovedEvent.java:9]: here i would keep passing the deleted absence object

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryRemovedEvent.java [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
 
-public class EmploymentEntryRemovedEvent extends AvpEvent {
+public class EmploymentEntryRemovedEvent extends AvpUpdatedEvent {
 
-    private final EmploymentEntry employment;
-
-    public EmploymentEntryRemovedEvent(Avp avp, EmploymentEntry employment) {
-        super(avp, new DomainEventType(EmploymentEntryRemovedEvent.class));
-        this.employment = employment;
+    public EmploymentEntryRemovedEvent(Avp avp, ChangeLog changeLog) {
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryRemovedEvent.java:9]: here i would keep passing the deleted object

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/InabilityEntryRemovedEvent.java [hunk 4]
```diff
@@ -1,19 +1,13 @@
...
 
-public class InabilityEntryRemovedEvent extends AvpEvent {
+public class InabilityEntryRemovedEvent extends AvpUpdatedEvent {
 
-    private final InabilityEntry inability;
-
-    public InabilityEntryRemovedEvent(Avp avp, InabilityEntry inability) {
-        super(avp, new DomainEventType(InabilityEntryRemovedEvent.class));
-        this.inability = inability;
+    public InabilityEntryRemovedEvent(Avp avp, ChangeLog changeLog) {
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/InabilityEntryRemovedEvent.java:9]: here i would keep passing the deleted object

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryUpdatedEvent.java [hunk 5]
```diff
@@ -1,19 +0,0 @@
-package ch.admin.seco.alv.service.onlineform.domain.avp.events;
-
-import ch.admin.seco.alv.service.onlineform.core.events.DomainEventType;
-import ch.admin.seco.alv.service.onlineform.domain.avp.Avp;
-import ch.admin.seco.alv.service.onlineform.domain.avp.EmploymentEntry;
-
-public class EmploymentEntryUpdatedEvent extends AvpEvent {
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryUpdatedEvent.java:7]: why was this deleted?

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/SelfEmploymentEntryRemovedEvent.java [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
+import ch.admin.seco.alv.service.onlineform.core.domain.ChangeLog;
 import ch.admin.seco.alv.service.onlineform.core.events.DomainEventType;
 import ch.admin.seco.alv.service.onlineform.domain.avp.Avp;
 
-public class SelfEmploymentEntryRemovedEvent extends AvpEvent {
+public class SelfEmploymentEntryRemovedEvent extends AvpUpdatedEvent {
 
-    public SelfEmploymentEntryRemovedEvent(Avp avp) {
-        super(avp, new DomainEventType(SelfEmploymentEntryRemovedEvent.class));
+    public SelfEmploymentEntryRemovedEvent(Avp avp, ChangeLog changeLog) {
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/SelfEmploymentEntryRemovedEvent.java:9]: here i would keep passing the deleted object

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryUpdatedEvent.java (hunk 5 above)
Comment (emina-str) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/EmploymentEntryUpdatedEvent.java:7]: we just have `EntryAddedEvent` and `EntryDeletedEvent` It didn't seem necessary to have even thrid one.

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/AbsenceEntryRemovedEvent.java (hunk 2 above)
Comment (emina-str) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/avp/events/AbsenceEntryRemovedEvent.java]: fixed
